### PR TITLE
Improvements to Rust version and various general cleanups

### PR DESCRIPTION
**Pull request by nazar-pc**
This resolves all but last TODO in Rust code (which is Windows support).

Changes that are not specific to Rust:
* `kind` on `RtpHeaderExtension` is no longer optional (it was only in the type system, if someone tried to not specify it, it wouldn't work anyway)
* `fecMechanisms` was removed from `RtpCapabilities` as it wasn't used anywhere
* `producerId` field in RTP observer messages for worker moved from `internal` into `data` field for consistency, this was one exception which annoyed me

I have also updated changelog files.

### CODE REVIEW

**Review by jmillan — APPROVED** [review]
💯

File: src/RtpParameters.ts [hunk 1]
```diff
@@ -104,7 +99,7 @@ export type RtpHeaderExtension =
 	 * Media kind. If empty string, it's valid for all kinds.
```
Comment (ibc) [inline src/RtpParameters.ts:104]: Description is wrong :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+### Unreleased
```
Comment (ibc) [inline CHANGELOG.md:3]: Empty line after title, please.

File: rust/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+### 0.8.0
```
Comment (ibc) [inline rust/CHANGELOG.md:3]: Same.

**Review by ibc — CHANGES_REQUESTED** [review]
LGTM just cosmetic changes requested. Also note that changes in the website docs are required. I can do that but it needs to be done before a new release since here we are doing "breaking" changes (even if no one will probably notice).

File: worker/src/RTC/Router.cpp [hunk 4]
```diff
@@ -494,14 +494,14 @@ namespace RTC
 		return rtpObserver;
 	}
 
-	RTC::Producer* Router::GetProducerFromInternal(json& internal) const
+	RTC::Producer* Router::GetProducerFromData(json& internal) const
 	{
 		MS_TRACE();
 
 		auto jsonProducerIdIt = internal.find("producerId");
```
Comment (ibc) [inline worker/src/RTC/Router.cpp]: This is wrong, right? I mean, it should be `data` (argument name).

Comment (ibc) [reply worker/src/RTC/Router.cpp]: so also in the hpp file

**Review by ibc — CHANGES_REQUESTED** [review]

File: worker/src/RTC/Router.cpp (hunk 4 above)
Comment (nazar-pc) [reply worker/src/RTC/Router.cpp]: You are right, sorry

**Review by ibc — APPROVED** [review]

**Comment by ibc** [comment]
Waiting for CI before merging. I'm also changing it in mediasoup-client (not sure yet if needed).

**Comment by nazar-pc** [comment]
Not required, but would be nice to align mediasoup-client similarly to this, yes.

**Comment by ibc** [comment]
Released in 3.7.15.

**Comment by nazar-pc** [comment]
Please create Rust tag for 0.8.0 too

**Comment by ibc** [comment]
Done

**Pull request merged**